services/QuillLMS/app/controllers/canvas_integration/lti_controller.rb

```diff
@@ -22,7 +23,22 @@ def launch_config
     def sso
       @canvas_instance_id = CanvasInstance.find_by(url: params[:canvas_instance_url])&.id
       @button_text = SSO_BUTTON_TEXT
-      @button_url = Auth::Canvas::ACCESS_PATH
+      @button_params = { canvas_instance_id: @canvas_instance_id }
+      @button_url = Auth::Canvas::OMNIAUTH_REQUEST_PATH
+    end
+
+    private def import_user
+      CanvasIntegration::UserImporter.run(
```

Comment:
Not a huge issue, but we might consider moving this to an async job 1) for performance reasons and 2) to ensure the login page in Canvas loads (if there's an error raised in this flow, the xml page would 500).

We would lose an assurance that the user is created before login, so if that's an issue we could leave it synchronous.

Reply:
My thought here is if there's an issue creating a User, we fail fast here and render some error for the user.  Putting it in a background job will push the error downstream.

I do still have to put a redirect for the error handling but that's pegged with a later PR.